presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1242,6 +1243,16 @@ private boolean canResolveOperator(OperatorType operatorType, List<TypeSignature
         }
     }
 
+    private ConnectorTableLayoutHandle getTableLayout(Session session, TableHandle tableHandle)
+    {
+        if (tableHandle.getLayout().isPresent()) {
+            return tableHandle.getLayout().get();
+        }
+        Optional<TableLayoutResult> result = getLayout(session, tableHandle, alwaysTrue(), Optional.empty());
```

Comment:
We could add some comments about what does it when `tableHandle.getLayout()` is empty.

Reply:
Actually looks like this get changed in later commits. And the final version is quite straightforward :)